dex/src/main/scala/com/wavesplatform/dex/db/AssetsCache.scala

```diff
@@ -0,0 +1,72 @@
+package com.wavesplatform.dex.db
+
+import cats.Id
+import com.wavesplatform.dex.domain.asset.Asset
+import com.wavesplatform.dex.domain.utils.ScorexLogging
+import com.wavesplatform.dex.grpc.integration.dto.BriefAssetDescription
+
+import java.util.concurrent.ConcurrentHashMap
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Success
+
+abstract class AssetsCache(implicit ec: ExecutionContext) {
+  val cached: AssetsDb[Id]
+
+  def contains(asset: Asset): Future[Boolean] = get(asset).map(_.nonEmpty)
+  def get(asset: Asset): Future[Option[BriefAssetDescription]]
+  def put(asset: Asset, item: BriefAssetDescription): Future[Unit]
+}
+
+object AssetsCache {
+
+  def from(storage: AssetsDb[Future])(implicit ec: ExecutionContext): AssetsCache =
+    new AssetsCache with ScorexLogging {
+      private val assetsCache = new ConcurrentHashMap[Asset, BriefAssetDescription]
+
+      override val cached = new AssetsDb[Id] {
+
+        override def put(asset: Asset.IssuedAsset, item: BriefAssetDescription): Unit =
+          throw new IllegalAccessException("This method should not be called, use AssetsCache.put!")
+
+        override def get(asset: Asset.IssuedAsset): Option[BriefAssetDescription] = Option(assetsCache.get(asset))
```

Comment:
can we just replace all invocations like ```assetsCache.cached.get``` with ```assetsCache.getUnsafe``` for example (which will take a value from assetsCache, trigger ```storage.get``` if needed and return immediately)? 
Anyway we can trigger ```storage.get``` here in case of required BriefAssetDescription is empty

Reply:
or consider replacing
```val cached: AssetsDb[Id]```
with
```val cached: SomeProperInterfaceWithoutPut[Id]```

Reply:
We will eliminate all unsafe calls during the refactoring. Fixed by the second approach right now.